x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/conversations/append_conversation_messages.ts

```diff
@@ -100,10 +100,14 @@ export const transformToUpdateScheme = (updatedAt: string, messages: Message[])
       is_error: message.isError,
       reader: message.reader,
       role: message.role,
-      trace_data: {
-        trace_id: message.traceData?.traceId,
-        transaction_id: message.traceData?.transactionId,
-      },
+      ...(message.traceData
+        ? {
+            trace_data: {
+              trace_id: message.traceData.traceId ?? '',
+              transaction_id: message.traceData.transactionId ?? '',
+            },
```

Comment:
```suggestion
            trace_data: {
              trace_id: message.traceData.traceId,
              transaction_id: message.traceData.transactionId,
            },
```

The contents of traceData are optional, so we can leave off the nullish coalescing. Right now this will make the 'view APM trace' action visible at all times since we check for presence of these fields vs empty. We could change the check, but I think the current schema makes most sense for absence of relevant trace data here.

<p align="center">
  <img width="700" src="https://github.com/elastic/kibana/assets/2946766/ad7dc914-60ef-43ad-9846-6c9e5c2c57e0" />
</p> 


Will need to update the other two occurrences from https://github.com/elastic/kibana/pull/174126/commits/8f0e7759285c385bf2a4acce9cf930842334dc0f over in create/update conversation as well.